src/builder/reduce.js

```diff
@@ -23,33 +23,37 @@ module.exports = function (file, ipld, options) {
         })
       }
 
-      // we are using raw leaf nodes, this file only has one node but it'll be marked raw
-      // so convert it back to a file node
+      // we're using raw leaf nodes so we convert the node into a UnixFS `file` node.
       return waterfall([
-        (cb) => ipld.get(new CID(leaf.multihash), cb),
+        (cb) => ipld.get(leaf.cid, cb),
         (result, cb) => {
-          const meta = UnixFS.unmarshal(result.value.data)
-          const fileNode = new UnixFS('file', meta.data)
+          const data = result.value.data
+          const fileNode = new UnixFS('file', data)
 
-          DAGNode.create(fileNode.marshal(), [], options.hashAlg, (err, node) => {
-            cb(err, { DAGNode: node, fileNode: fileNode })
+          DAGNode.create(fileNode.marshal(), [], options.hashAlg, (error, node) => {
+            cb(error, { DAGNode: node, fileNode: fileNode })
```

Comment:
Nitpick, why s/err/error?

Reply:
Personal preference really - I think short variable names make the code less expressive.